tests/test_advanced_options.py

```diff
@@ -99,3 +101,24 @@ def test_analysis_of_private_repo(analysis_data):
     assert len(violations) > 1, "Expected issues are missing";
 
     manage_credentials_in_maven_xml(custom_maven_settings, True)
+
+def test_no_container_leftovers(analysis_data):
+    application_data = analysis_data['jee_example_app']
+    command = build_analysis_command(
+        application_data['file_name'],
+        application_data['source'],
+        application_data['target']
+    )
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
+
+    time.sleep(10)
+    process.send_signal(signal.SIGINT)
+
+    process.wait()
+
+    leftover = subprocess.run("podman ps", shell=True, stdout=subprocess.PIPE, encoding='utf-8')
+    leftover_output = leftover.stdout.strip()
+
+    for line in leftover_output.splitlines():
+        assert not line.startswith("analysis-"), f"Found a leftover container with name: {line}"
+        assert not line.startswith("provider-"), f"Found a leftover container with name: {line}"
```

Comment:
When we run `podman ps`, the lines it returns start always with the ID of the container, so no line will start with a word, hence the test will always pass:

```bash
CONTAINER ID  IMAGE                                                             COMMAND       CREATED       STATUS       PORTS       NAMES
b14ee9e791ed  registry.redhat.io/mta/mta-generic-external-provider-rhel9:7.1.1  --port=42063  1 second ago  Up 1 second              provider-IOEBjofLLGdoYqvD

```